clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerMetadata.java

```diff
@@ -46,23 +47,26 @@ public ProducerMetadata(long refreshBackoffMs,
         LogContext logContext,
         ClusterResourceListeners clusterResourceListeners,
         Time time) {
-        this(refreshBackoffMs, metadataExpireMs, logContext, clusterResourceListeners, time, TOPIC_EXPIRY_MS);
+        this(refreshBackoffMs, metadataExpireMs, logContext, clusterResourceListeners, time, TOPIC_EXPIRY_MS, true);
```

Comment:
Why are we passing true here when the default is false?

Reply:
this constructor is only used for test case. Let me add a comment.

Reply:
can you mention the hotfix that this pertains to as well?

Reply:
added comments.